cms/models.py

```diff
@@ -0,0 +1,65 @@
+from django.db import models
+from django.conf import settings
+from ckeditor_uploader.fields import RichTextUploadingField
+import pathlib
+
+
+def load_template_choices():
+    return [
+        (f.name, f.name)
+        for f in (pathlib.Path(__file__).parent / "templates/").iterdir()
+        if f.is_file()
+    ]
+
+
+class BasePage(models.Model):
+    title = models.CharField("Titel", max_length=100)
+    template = models.CharField(
+        "Template", max_length=100, choices=[("", "")], blank=False, default=None
+    )
+    content = RichTextUploadingField(blank=True)
+
+    class Meta:
+        abstract = True
+        verbose_name = "Basis Seite"
+        verbose_name_plural = verbose_name
+
+    def __init__(self, *args, **kwargs):
+        self._meta.get_field("template").choices = load_template_choices()
+        super(BasePage, self).__init__(*args, **kwargs)
+
+
+class Menu(models.Model):
+    name = models.CharField("Name", max_length=15)
+
+    class Meta:
+        verbose_name = "Menü"
+        verbose_name_plural = "Menüs"
+
+    def __str__(self):
+        return self.name
+
+
+class StaticMenuPage(BasePage):
+    menu = models.ForeignKey(
+        Menu,
+        on_delete=models.PROTECT,
```

Comment:
It should be possible to not assign a menu: make nullable and cascate with `models.SET_NULL`